File: client/iih_stub/iih.ci_stub/iih.ci.mr/nu/obstetrics/opernurecord/d/OpernurecordAggDO.cs
using System;
using xap.mw.core.data;
using xap.mw.core.meta;

namespace iih.ci.mr.nu.obstetrics.opernurecord.d
{
	/// <summary>
	/// 手术护理记录单DO
	/// </summary>
	public class OpernurecordAggDO : BaseAggDO
	{	
	    private static string OPERNURECORDDO_OPERNURECORDEQMDO="iih.ci.mr.nu.obstetrics.opernurecord.d.OperNuRecordEqmDO";
	    private static string OPERNURECORDDO_OPERNURECORDDRESSDO="iih.ci.mr.nu.obstetrics.opernurecord.d.OperNuRecordDressDO";

        public OpernurecordAggDO() {
            this.setParentDO(new OperNuRecordDO());
        }

  	    public OperNuRecordDO getParentDO() {
		    return ((OperNuRecordDO) base.ParentDO);
	    }      
	    
	    public void setParentDO(OperNuRecordDO headDO) {
		    base.ParentDO=headDO;
	    }
	    
	    public OperNuRecordEqmDO[] getOperNuRecordEqmDO() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(OPERNURECORDDO_OPERNURECORDEQMDO));
		    if (dos == null || dos.Length==0){return null;}
            OperNuRecordEqmDO[] r = new OperNuRecordEqmDO[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as OperNuRecordEqmDO;
            }
	        return r;
		    
	    }
	    
	    public void setOperNuRecordEqmDO(OperNuRecordEqmDO[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }
	    public OperNuRecordDressDO[] getOperNuRecordDressDO() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(OPERNURECORDDO_OPERNURECORDDRESSDO));
		    if (dos == null || dos.Length==0){return null;}
            OperNuRecordDressDO[] r = new OperNuRecordDressDO[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as OperNuRecordDressDO;
            }
	        return r;
		    
	    }
	    
	    public void setOperNuRecordDressDO(OperNuRecordDressDO[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }


	    public override BaseDO createParentDO()
	    {
	        return new OperNuRecordDO();
	    }

        public override BaseDO createChildDO(string clzName)
	    {
            if (OPERNURECORDDO_OPERNURECORDEQMDO == clzName)
	        {
                return new OperNuRecordEqmDO();
	        }
           else if (OPERNURECORDDO_OPERNURECORDDRESSDO == clzName)
	        {
                return new OperNuRecordDressDO();
	        }
	        return null;
	    }
	    
	    /// <summary>
        /// 获取子类全路径名数组
        /// </summary>
        /// <returns></returns>
        public override string[] getChildClassNames(){
            return  new string[]{"iih.ci.mr.nu.obstetrics.opernurecord.d.OperNuRecordEqmDO","iih.ci.mr.nu.obstetrics.opernurecord.d.OperNuRecordDressDO"};
        }
	    public new static AggMeta GetMeta()
	    {
	        AggMeta meta = new AggMeta();
	        meta.OneToOneDOs = new string[]{};
	        return meta;
	    }
	    
	}
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/mr/nu/obstetrics/opernurecord/d/OpernurecordAggDO.java
package iih.ci.mr.nu.obstetrics.opernurecord.d;

import xap.sys.appfw.orm.model.agg.BaseAggDO;
import xap.mw.core.data.BaseDO;
import xap.mw.coreitf.i.IAggDesc;
import xap.mw.coreitf.i.IBaseDO;
import iih.ci.mr.nu.obstetrics.opernurecord.d.desc.OpernurecordAggDODesc;



/**
 * 手术护理记录单
 */
public class OpernurecordAggDO extends BaseAggDO {
	private static final long serialVersionUID = 1L;

	@Override
	public OperNuRecordDO getParentDO() {
		return ((OperNuRecordDO) super.getParentDO());
	}

	public void setParentDO(OperNuRecordDO headDO) {
		setParent(headDO);
	}

	public OperNuRecordEqmDO[] getOperNuRecordEqmDO() {
		IBaseDO[] dos = getChildren(OperNuRecordEqmDO.class);
		if(dos==null) return null;
		OperNuRecordEqmDO[] result=new OperNuRecordEqmDO[dos.length];
		for(int i=0;i<dos.length;i++){
			result[i]=(OperNuRecordEqmDO)dos[i];
		}
		return result;
	}
	
	public void setOperNuRecordEqmDO(OperNuRecordEqmDO[] dos) {
		setChildren(OperNuRecordEqmDO.class, dos);
	}
	public OperNuRecordDressDO[] getOperNuRecordDressDO() {
		IBaseDO[] dos = getChildren(OperNuRecordDressDO.class);
		if(dos==null) return null;
		OperNuRecordDressDO[] result=new OperNuRecordDressDO[dos.length];
		for(int i=0;i<dos.length;i++){
			result[i]=(OperNuRecordDressDO)dos[i];
		}
		return result;
	}
	
	public void setOperNuRecordDressDO(OperNuRecordDressDO[] dos) {
		setChildren(OperNuRecordDressDO.class, dos);
	}

	@Override
	public IAggDesc getAggDesc() {
		OpernurecordAggDODesc desc = new OpernurecordAggDODesc();

		return desc;
	}
	
	 public  BaseDO createParentDO() {
         return new OperNuRecordDO();
     }

     public  BaseDO createChildDO(String clzName) {
	     if (clzName.equals("iih.ci.mr.nu.obstetrics.opernurecord.d.OperNuRecordEqmDO")) {
             return new OperNuRecordEqmDO();
         }
	    else if (clzName.equals("iih.ci.mr.nu.obstetrics.opernurecord.d.OperNuRecordDressDO")) {
             return new OperNuRecordDressDO();
         }
         return null;
     }
}